samza-core/src/main/java/org/apache/samza/table/remote/RemoteTable.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.table.ReadableTable;
+import org.apache.samza.task.TaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Remote store backed readable table
+ *
+ * @param <K> the type of the key in this table
+ * @param <V> the type of the value in this table
+ */
+public class RemoteTable<K, V> implements ReadableTable<K, V> {
+  static final String READER_DATA = "io.reader";
+  static final String WRITER_DATA = "io.writer";
+
+  protected String tableId;
+  protected Logger logger;
+  protected RecordReader<K, V> recordReader;
+
+  public RemoteTable(String tableId, RecordReader recordReader) {
+    Validate.notBlank(tableId, "invalid table id");
+    Validate.notNull(recordReader, "null record reader");
+    this.tableId = tableId;
+    this.recordReader = recordReader;
+    this.logger = LoggerFactory.getLogger("RemoteTable-" + tableId);
```

Comment:
this is per-tableId, so cannot be static. I think there is value in having the log messages be clear on which table instance it is from, right? Otherwise, we would need to append "tableId={}" to all the log statements.